src/helpers.ts

```diff
@@ -76,167 +69,37 @@ export async function createAsset(
 	providerUrl: string,
 	config: Config,
 	aquariusInstance: Aquarius,
+	encryptDDO: boolean = true,
 	templateIndex: number = 1,
+	providerFeeToken: string = ZERO_ADDRESS,
+	accessListFactory?: string,
+	allowAccessList?: string,
+	denyAccessList?: string,
 	macOsProviderUrl?: string,
-	encryptDDO: boolean = true,
+	
 ) {
 	const { chainId } = await owner.provider.getNetwork();
-	const nft = new Nft(owner, chainId);
-	const nftFactory = new NftFactory(config.nftFactoryAddress, owner);
-
-	let wrappedSigner
-	let allowListAddress
 	if(templateIndex === 4){
 		// Wrap the signer for Sapphire
-		wrappedSigner = sapphire.wrap(owner);
+		const wrappedSigner = sapphire.wrap(owner);
 
 		// Create Access List Factory
-		const accessListFactory = new AccesslistFactory(config.accessListFactory, wrappedSigner, chainId);
+		const accessListFactoryObj = new AccesslistFactory(config.accessListFactory, wrappedSigner, chainId);
 
 		// Create Allow List
-		allowListAddress = await accessListFactory.deployAccessListContract(
+		await accessListFactoryObj.deployAccessListContract(
```

Comment:
are we sure about these params?
`config.accessListFactory `
`allowAccessList`
`denyAccessList`
if there do not match probably there will be errors at smart contract level (not sure if console.log will help)
btw, there is a function to deploy the access list on the SDK as well
`export async function createAccessListFactory(`
we could probably just use that one instead

Reply:
As far as I double-checked, I put them in place to match createAsset from ocean.js

Reply:
its not about the order of the params:
`accessListFactoryObj`, `allowAccessList` and `denyAccessList` should be contract addresses (strings)

Here, you're creating a new access list factory and a new access list from that factory
```
// Create Access List Factory
		const accessListFactoryObj = new AccesslistFactory(config.accessListFactory, wrappedSigner, chainId);

		// Create Allow List
		await accessListFactoryObj.deployAccessListContract(
			'AllowList',
			'ALLOW',
			['https://oceanprotocol.com/nft/'],
			false,
			await owner.getAddress(),
			[await owner.getAddress(), ZERO_ADDRESS]
		)
		return await createAsset(name, symbol, wrappedSigner, assetUrl, templateIndex, ddo, encryptDDO, providerUrl || macOsProviderUrl, providerFeeToken, aquariusInstance, accessListFactory, allowAccessList, denyAccessList)
```
But you're calling the function with other values, the factory addresses and the allow access list addresses do not match (you're creating new ones and passing something else)